tests/vmi_lifecycle_test.go

```diff
@@ -725,6 +725,146 @@ var _ = Describe("[rfe_id:273][crit:high][arm64][vendor:cnv-qe@redhat.com][level
+				vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Create(vmi)
+				Expect(err).ToNot(HaveOccurred(), "Should create VMI successfully")
+
+				// Ensure that the VMI is running. This is necessary to ensure that virt-handler is fully responsible for
+				// the VMI to build the original cache.
+				nodeName := tests.WaitForSuccessfulVMIStart(vmi)
+
+				virtHandler, err = kubecli.NewVirtHandlerClient(virtClient).Namespace(flags.KubeVirtInstallNamespace).ForNode(nodeName).Pod()
+				Expect(err).ToNot(HaveOccurred(), "Should get virthandler client")
+
+				ds, err := virtClient.AppsV1().DaemonSets(virtHandler.Namespace).Get(context.Background(), "virt-handler", metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred(), "Should get virthandler daemonset")
+				// Save virt-handler number of desired pods
+				virtHandlerAvailablePods = ds.Status.DesiredNumberScheduled
+
+				kv := util.GetCurrentKv(virtClient)
+				kv.Spec.Workloads = &v1.ComponentConfig{
+					NodePlacement: &v1.NodePlacement{
+						Affinity: &k8sv1.Affinity{
+							NodeAffinity: &k8sv1.NodeAffinity{
+								RequiredDuringSchedulingIgnoredDuringExecution: &k8sv1.NodeSelector{
+									NodeSelectorTerms: []k8sv1.NodeSelectorTerm{
+										{MatchExpressions: []k8sv1.NodeSelectorRequirement{
+											{Key: "kubernetes.io/hostname", Operator: "NotIn", Values: []string{nodeName}},
+										}},
+									},
+								},
+							},
+						},
+					},
+				}
+				_, err = virtClient.KubeVirt(kv.Namespace).Update(kv)
+				Expect(err).ToNot(HaveOccurred(), "Should update kubevirt infra placement")
+
+				Eventually(func() bool {
+					_, err := virtClient.CoreV1().Pods(virtHandler.Namespace).Get(context.Background(), virtHandler.Name, metav1.GetOptions{})
+					return errors.IsNotFound(err)
+				}, 200*time.Second, 1*time.Second).Should(BeTrue(), "The virthandler pod should be gone")
+			})
+
+			It("[test_id:1652]needs to get latest resourceVersion for updating vmi", func() {
+
+				oldVMI, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+				resourceVersion := oldVMI.ResourceVersion
+				Expect(err).To(BeNil(), "Should get vmi successfully")
+				Expect(resourceVersion).ToNot(BeEmpty(), "Should be assigned with an resource Version")
+
+				By("Killing the VirtualMachineInstance")
+				err = pkillAllVMIs(virtClient, nodeName)
+				Expect(err).To(BeNil(), "Should kill qemu process by killer pod successfully")
+
+				curVMI := oldVMI.DeepCopy()
+				curVMI.Status.ActivePods = nil
+				updatedVMI, err := virtClient.VirtualMachineInstance(vmi.Namespace).Update(curVMI)
+				Expect(err).To(BeNil(), "Should update vmi successfully with same resourceVersion")
+				Expect(updatedVMI.ResourceVersion).NotTo(Equal(resourceVersion), "Should be assigned with a new resource Version")
+				resourceVersion = updatedVMI.ResourceVersion
```

Comment:
Do you apply this change?  I don't see an `.Update` after ResourceVersion is changed

Reply:
original RV is from oldVMI.ResourceVersion (Line780). First `.Update` object (Line790), so the updatedVMI.ResourceVersion should not equal to original one. All these operations are executed while virt-handler was gone.
Then restore virt-handler to simulate virt-handler recovered from crash or reboot. It should rebuild informer cache, which automatically triggers UpdateFunc, so new resourceVersion will be get as we expected to fix.
Line814 checks the latest resourceVersion should not equal to updatedVMI.ResourceVersion (Line793).